File: ALiPayServer/src/main/java/com/scrat/server/gogo/dao/UserCoinDao.java
package com.scrat.server.gogo.dao;

import com.scrat.server.gogo.base.BaseDao;
import com.scrat.server.gogo.base.BaseRowObj;
import org.springframework.stereotype.Repository;

import java.util.Date;

/**
 * Created by scrat on 2017/11/7.
 */
@Repository
public class UserCoinDao extends BaseDao {
    public BaseRowObj getUserCoin(String uid) {
        String sql = "select * from user_coin where uid=? limit 1";
        return queryForMap(sql, uid);
    }

    public long addUserCoin(String uid, int count) {
        long nowTs = new Date().getTime();
        String sql = "insert ignore into user_coin set uid=?,coin_count=?,create_ts=?";
        return insert(sql, "user_coin_id", uid, count, nowTs);
    }

    public void updateUserCoin(String uid, int count) {
        long nowTs = new Date().getTime();
        String sql = "update user_coin set coin_count=?,update_ts=? where uid=?";
        update(sql, count, nowTs, uid);
    }
}

File: WechatPayServer/dao/UserCoinDao.js
function * findByUid(uid) {
    return yield Conn.query(
        'select * from user_coin where uid=? limit 1', {replacements: [uid], type: Sequelize.QueryTypes.SELECT});
}

function * addCoin(uid, coin) {
    yield Conn.query('update user_coin set coin_count=coin_count+? where uid=?',
        {replacements: [coin, uid], type: Sequelize.QueryTypes.UPDATE});
}

function * addUserCoin(uid, coin_count) {
    yield Conn.query('insert ignore into user_coin set uid=?,coin_count=?,create_ts=?',
        {replacements: [uid, coin_count, new Date().getTime()], type: Sequelize.QueryTypes.INSERT});
}

module.exports = {
    findByUid, addCoin, addUserCoin
};